core/spec/models/spree/order_contents_spec.rb

```diff
@@ -243,6 +292,19 @@
       }.to change { subject.order.total }
     end
 
+    context 'when store_credits payment' do
+      let!(:payment) { create(:store_credit_payment, order: order) }
+
+      it 'is expected to have store_credit_payment' do
+        expect(order.payments.store_credits).to include(payment)
+      end
+
+      it "is expected to delete store_credits payment" do
+        subject.update_cart params
+        expect(order.payments.store_credits.count).to eq(0)
```

Comment:
It would be nicer to use RSpec [change matcher](https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher) :)